Chocobo Circuit grandstand exit NPC's

Tom-Neverwinter opened the pull request:
please formatting gods

/makes donation at lua temple

/burns efugy

<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [X] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [X] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

adds chocobo circuit grandstand exit NPC's

## Steps to test these changes

<!-- Clear and detailed steps to test your changes here -->

!pos and test, make sure exits to just outside the entrance counterpart.

Tom-Neverwinter committed:
Chocobo Circuit grandstand exit NPC's

please formatting gods

/makes donation at lua temple

/burns efugy

zach2good commented:
From CI:
```
Invalid multiline conditional format: scripts/zones/Chocobo_Circuit/npcs/Azainnie.lua:19
if csid == 328 and option == 1                              <-- HERE

...

Multiple parameters used without an appropriate following space or newline: scripts/zones/Chocobo_Circuit/npcs/Odersille.lua:21:33
player:setpos(-251.1373,-5.000,-489.9807) -- exits to saffa                              <-- HERE

...

Could not find function match for setpos (./scripts/zones/Chocobo_Circuit/npcs/Flige.lua:21:16)
Could not find function match for setpos (./scripts/zones/Chocobo_Circuit/npcs/Perdric.lua:21:16)
Could not find function match for setpos (./scripts/zones/Chocobo_Circuit/npcs/Odersille.lua:21:16)
Could not find function match for setpos (./scripts/zones/Chocobo_Circuit/npcs/Azainnie.lua:21:16)
Could not find function match for setpos (./scripts/zones/Chocobo_Circuit/npcs/Couzanne.lua:21:16)
``` 

> Could not find function match for setpos

_Please_ test your code, you checked the box. This will have never worked as functions are case-sensitive.

> General formatting of if-statements

https://github.com/LandSandBoat/server/wiki/Development-Guide-Lua#formatting-conditional-blocks

Example:

```lua
entity.onEventFinsih = function(player, csid, option)
    if csid == 328 and option == 1 then
        player:setPos(-330.231, -5.000, -413.392) -- exits to illsoire
    end
end
```

The rest of the codebase has been slowly reformatted over the last couple of months, so it should serve as a good example of how to lay things out (previously it hasn't, but Clay has put in lots of manual labour to make it look clean, uniform, and understandable).

zach2good commented:
If you're unsure about the lua formatting and don't want to continually bounce your work off of CI, you can run the scripts locally: 
`./tools/ci/lua.sh`

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
changes as per WinterSolstice8

test 1

Xaver-DaRed commented on scripts/zones/Chocobo_Circuit/npcs/Azainnie.lua:
```diff
@@ -0,0 +1,27 @@
+end
+
+entity.onEventFinsih = function(player, csid, option)
+if
```
this "if" (and others below) is missalligned

zach2good replied:
Yeah, it's unfortunate that CI didn't/doesn't catch this. Available Lua formatters are very very basic, so we're essentially building our own over time. The best thing you can do until our formatted is bulletproof is compare your code to other files in the codebase and see if it matches up, the rest of the Lua code is in fairly good shape these days

Tom-Neverwinter committed:
change if alignment

adjusted if alignment based on multiple other npc scripts

Tom-Neverwinter committed:
Copy from correct directory

Tom-Neverwinter committed:
chocobo_Circuit Grandstand entrance NPCs

--author cocosolos  cocosolos@users.noreply.github.com

Tom-Neverwinter committed:
Formatting changes

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

cocosolos commented on scripts/zones/Chocobo_Circuit/npcs/Couzanne.lua:
```diff
@@ -0,0 +1,27 @@
+entity.onTrade = function(player, npc, trade)
+end
+
+entity.onTrigger = function(player, npc)
+    player:startEvent(332)
+end
+
+entity.onEventUpdate = function(player, csid, option)
+end
+
+entity.onEventFinsih = function(player, csid, option)
+    if
+        csid == 332 and
+        option == 1
+    then
+        player:setPos(-387.2634, -5.000, -490.0573) -- exits to Biorie
+    end
+end
```
For these exit NPCs you shouldn't be setting pos or doing anything in the onEventFinish. Instead when you start the cutscene you should add whatever the corresponding parameter is for the entry NPC. For example with Couzanne you would do `player:startEvent(332, 5)` because Biorie uses 5 and these 2 NPCs are linked. Likewise Azainnie would use `player:startEvent(328, 1)` because Ilsoire uses 1 etc.

Tom-Neverwinter replied:
I always end up at the same npc when I use 1-5 as paramater 2

cocosolos replied:
I would double check that you have nothing in the onEventFinish and that the startEvent parameters are correct. I've tested and confirmed that `player:startEvent(csid, entry)` is the correct format for the exit NPCs and takes you to the correct place depending on the entry parameter. Note that the 1-5 index comes immediately after the csid unlike the entry NPCs. Example:
```lua
-----------------------------------
-- Area: Chocobo_Circuit
-- NPC: Perdric
-- !pos -84.007 -14.500 -133.451 70
-----------------------------------
local entity = {}

entity.onTrade = function(player, npc, trade)
end

entity.onTrigger = function(player, npc)
    player:startEvent(330, 3)
end

entity.onEventUpdate = function(player, csid, option)
end

entity.onEventFinsih = function(player, csid, option)
end

return entity
```

Tom-Neverwinter committed:
a

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
add note

noted npc for in and out

Tom-Neverwinter committed:
0123456789 not descriptive enough so heres a gate

Tom-Neverwinter committed:
Delete the gate to rerun CI

not sure if ci is checking to see if I have comments too...

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Update npc see if I figured out better formatting

formatting fix

Tom-Neverwinter committed:
Update NPC based on feedback

The original code had several issues:

Multiple parameters used without an appropriate following space or newline
Semicolon detected in line
No newlines before end statement
Could not find function match for setpos
I have corrected these issues by:

Adding appropriate spaces between the parameters for the setpos function.
Removing the semicolon after the end statement
Adding a newline before the end statement
Replacing setpos with setPos.

Tom-Neverwinter committed:
another test

I want to know if this changed anything

Tom-Neverwinter committed:
title length, add comment

Test title length ci issue and add comment for exit point

Tom-Neverwinter committed:
Chocobo betting attendants

First two of the five? sets

Tom-Neverwinter committed:
Chocobo betting attendants 2

add more betting attendants

Tom-Neverwinter committed:
Chocobo betting attendants 3

should be all of them

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
fix spelling

Tom-Neverwinter committed:
incomplete give player map on interact in chocobo circuit

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
spelling updates

map still needs proper id

Tom-Neverwinter committed:
spaces fix

Tom-Neverwinter committed:
try and fix line endings [dos2unix-7.4.4-win64]

dos2unix-7.4.4-win64 [ai suggested, github clearly never touches and changes the line endings...]

Tom-Neverwinter committed:
fix format

spaces

Tom-Neverwinter committed:
delete

Tom-Neverwinter committed:
add proper code with proper ending

whoops. didnt change .txt to .lua

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Minimum length is 10 characters.

Minimum length is 10 characters.

Tom-Neverwinter committed:
delete to resolve Minimum length is 10 characters.

Delete file to bypass Minimum length is 10 characters.

should make it finally pass

zach2good commented:
You need to rebase, currently you're just adding more commits instead of fixing the ones you've already got

zach2good commented:
https://git-scm.com/book/en/v2/Git-Branching-Rebasing

zach2good commented:
> it wont update or pass because it apparently applies to any commit in the chain

This is intentional. If we took your set of commits as-is, that short message of "delete" would come with it. For people looking at these commits in the future: what are you deleting? How much information does "delete" convey?

Like here at a glance, I can see the bit of work that this line was last modified by:
<img width="494" alt="image" src="https://user-images.githubusercontent.com/1389729/226852244-0ac39a28-f251-4882-8d5f-50c47dfd50ed.png">

zach2good commented on 00000.lua:
```diff
@@ -0,0 +1,30 @@
+entity.onEventFinish = function(player, csid, option)
+end
+
+return entity
```
GitHub is reporting this as not having newlines at the end of the file, but CI hasn't caught it for some reason 🤔

All files need to end with a newline

zach2good commented on Boirie.lua:
```diff
@@ -0,0 +1,31 @@
+entity.onEventFinish = function(player, csid, option)
+end
+
+return entity
```
Newline

zach2good commented on 00000.lua:
```diff
@@ -0,0 +1,30 @@
+-----------------------------------
```
A bunch of these files are in the root of the repo. They obviously shouldn't be here

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Cadwyn.lua:
```diff
@@ -0,0 +1,21 @@
+end
+
+entity.onTrigger = function(player, npc)
+player:startEvent(212, 213, 214, 215, 218, 222, 228, 233)
```
This isn't formatted correctly

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Flige.lua:
```diff
@@ -0,0 +1,21 @@
+end
+
+entity.onTrigger = function(player, npc)
+    player:startEvent(329, 2) --Foulneporde
```
Comment is formatted incorrectly

zach2good reviewed (changes requested):
For whatever reason, it looks like some portions of CI aren't running, or this branch is so old that it doesn't have the most recent CI updates. You need to rebase to a) fix your git commits, and b) get CI to work properly on this PR

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Sudha Rhilimanyme.lua:
```diff
@@ -0,0 +1,21 @@
+-----------------------------------
+-- Area: Chocobo_Circuit
```
Theres a space in the filename here, it should be an underscore

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Fhaht Beablrucch.lua:
```diff
@@ -0,0 +1,21 @@
+-----------------------------------
```
Theres a space in the filename here, it should be an underscore

zach2good replied:
Presumably, that means this doesn't even work, depending on whats in the db

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

zach2good commented:
You've closed the PR now, so the big buttons that easily take you to the results of the latest CI run aren't there anymore, but it's still easily possible to click on the little red cross or green tick next to each commit:
<img width="649" alt="image" src="https://user-images.githubusercontent.com/1389729/227889414-e565b5e4-2b8d-4c26-bf47-156a79aaa5a7.png">

and it'll take you to the CI run:
https://github.com/LandSandBoat/server/actions/runs/4527736935/jobs/7973865995

Where you can see Sanity Checks > Git Formatting Checks has failed:
<img width="548" alt="image" src="https://user-images.githubusercontent.com/1389729/227889750-48f1fb58-34c4-462e-ab24-8465b01a0577.png">

With the following details:
```
Run touch git_checks.txt
### Found formatting issue in commit ###
3586b4b4a6a867ee256dc5d7e46140ebfc8e54a3

### Issue ###
Detected unhelpful language in commit message.
Please describe what you have changed with this commit.
Unhelpful language:
Oops
Whoops
lol

### Line with issue ###
whoops. didnt change .txt to .lua

### Commit message ###
add proper code with proper ending
whoops. didnt change .txt to .lua

### Found formatting issue in commit ###
2ecb865ff83d5688db7265e1e041bb79[11](https://github.com/LandSandBoat/server/actions/runs/4527736935/jobs/7973865995#step:6:12)4f3d3e

### Issue ###
Commit title is too short.
Minimum length is 10 characters.
Please provide a more detailed summary of your changes.

### Line with issue ###
delete

### Commit message ###
delete

### Found formatting issue in commit ###
0870755d9bd614e418b240000266cf7d1046afb1

### Issue ###
Commit title is too short.
Minimum length is 10 characters.
Please provide a more detailed summary of your changes.

### Line with issue ###
add note

### Commit message ###
add note
noted npc for in and out

### Found formatting issue in commit ###
c9b[13](https://github.com/LandSandBoat/server/actions/runs/4527736935/jobs/7973865995#step:6:14)2f18d33bd87c155a[14](https://github.com/LandSandBoat/server/actions/runs/4527736935/jobs/7973865995#step:6:15)38b7[19](https://github.com/LandSandBoat/server/actions/runs/4527736935/jobs/7973865995#step:6:20)d7babd8e7af

### Issue ###
Commit title is too short.
Minimum length is 10 characters.
Please provide a more detailed summary of your changes.

### Line with issue ###
a

### Commit message ###
a
```
>  I probably need to make changes however the output on ci isnt helpful or reliable.

It tells you exactly what needs to be done, and I've already linked you to tutorials on how to rebase/rename/edit/squash your commits to give them better names (link to that comment here: https://github.com/LandSandBoat/server/pull/3202#issuecomment-1479140778). As for reliability, the sanity checks and builds are rock-solid. The startup checks can sometimes get a little flakey due to port contention on the GitHub shared CI machines, but this PR isn't even making it to that step. 

As the log says:
```
### Issue ###
Commit title is too short.
Minimum length is 10 characters.
Please provide a more detailed summary of your changes.

### Line with issue ###
a
```

```
### Found formatting issue in commit ###
3586b4b4a6a867ee256dc5d7e46140ebfc8e54a3

### Issue ###
Detected unhelpful language in commit message.
Please describe what you have changed with this commit.
Unhelpful language:
Oops
Whoops
lol

### Line with issue ###
whoops. didnt change .txt to .lua

### Commit message ###
add proper code with proper ending
whoops. didnt change .txt to .lua
```

etc.

zach2good commented:
My general workflow when I'm tinkering on something is to make my changes, stage them, make an initial commit, and then continually use `git commit --amend`, which allows me to combine my new changes with the changes from the first commit, and always be prompted to update the commit message as I go.

If I've already pushed and have a PR open, then I just push with `git push --force` and the new commit will replace the old commit in the PR (GitHub tracks all the changes in between, so there's little chance of accidentally obliterating your work if you do it wrong - provided you've pushed your good changes at least once).

zach2good commented:
You can see what that ends up looking like here:
<img width="656" alt="image" src="https://user-images.githubusercontent.com/1389729/227894291-cbf01699-c8a1-4709-82a8-17b529f17c16.png">

Its possible to inspect the before/after force push by clicking on the commit hashes
